cosmos/estimategas.go

```diff
@@ -0,0 +1,42 @@
+package cosmos
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+)
+
+// RegisterEstimateGasRoute regiters the route on the router.
+func RegisterEstimateGasRoute(cliCtx context.CLIContext, r *mux.Router) {
+	r.HandleFunc("/txs/estimategas", EstimateGasRequestHandlerFn(cliCtx)).Methods("POST")
```

Comment:
Let's try to keep the same naming than in cosmos, they usually call that `simulate`

```suggestion
	r.HandleFunc("/txs/simulate", EstimateGasRequestHandlerFn(cliCtx)).Methods("POST")
```